libbeat/cfgfile/list.go

```diff
@@ -226,3 +236,16 @@ func createRunner(factory RunnerFactory, pipeline beat.PipelineConnector, cfg *r
 	c, _ := config.NewConfigFrom(cfg.Config)
 	return factory.Create(pipetool.WithDynamicFields(pipeline, cfg.Meta), c)
 }
+
+type UnitError struct {
+	UnitID string
+	Err    error
+}
+
+func (u UnitError) Error() string {
+	return u.Err.Error()
```

Comment:
Would it be helpful to include `u.UnitID` (if it's not empty) in the returned string?

Reply:
Not realy, a Unit can have multiple inputs, so it's not even a direct mapping also when the error is reported by the Elastic-Agent it is already under the context of an input, so it makes the information redundant and the error text longer.

I created this type only to be able to "bubble up" the error linked to the Unit so the ManagerV2 can properly handle it, I also wanted to keep the final error message untouched.